### PR TITLE
feat(localization-messages): add autoware_localization_msgs

### PR DESCRIPTION
**Pull request by xmfcx**
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@xmfcx 
LGTM
Some comments :pray: 
- Do you know where is discussion localization msgs?  I'd like to refer to it in this PR.
- Now, we use the tier4 repository and clone it. If we merged this PR, there are no conflicts because they are different package names. But some developers will confuse `autoware_auto_localization_msgs` and `autoware_localization_msgs`. Do you know the plan for how to migrate this package? cc @mitsudome-r @kenji-miyake 
https://github.com/autowarefoundation/autoware/blob/main/autoware.repos#L5

**Comment by yukkysaito** [comment]
>Now, we use the tier4 repository and clone it. If we merged this PR, there are no conflicts because they are different package names. But some developers will confuse about autoware_auto_localization_msgs and autoware_localization_msgs. Do you know the plan for how to migrate this package? cc @mitsudome-r @kenji-miyake

Sorry, there was no package(autoware_auto_localization_msgs) about localization_msgs in tier4 repository. But we may need to plan to implement universe with this package

**Comment by xmfcx** [comment]
@yukkysaito wrote:
> * Do you know where is discussion localization msgs?  I'd like to refer to it in this PR.

Here: https://github.com/orgs/autowarefoundation/discussions/1

> * Now, we use the tier4 repository and clone it. If we merged this PR, there are no conflicts because they are different package names. But some developers will confuse about `autoware_auto_localization_msgs` and `autoware_localization_msgs`. Do you know the plan for how to migrate this package? cc @mitsudome-r @kenji-miyake
>   https://github.com/autowarefoundation/autoware/blob/main/autoware.repos#L5

autoware_auto_msgs will be removed. Right now it is in the `src/core/tmp/autoware_auto_msgs` temporary folder.

Also `autoware_auto_localization_msgs` doesn't exist.

File: autoware_localization_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.5)
+project(autoware_localization_msgs)
+
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
+
```
Comment (xmfcx) [inline autoware_localization_msgs/CMakeLists.txt:6]:
```suggestion
#test
```

**Pull request merged**